router/http_test.go

```diff
@@ -279,7 +279,9 @@ func httpHeaderTestHandler(c *C, ip, port string) http.Handler {
 		c.Assert(req.Header["X-Forwarded-Proto"][0], Equals, "http")
 		c.Assert(len(req.Header["X-Request-Start"][0]), Equals, 13)
 		c.Assert(req.Header["X-Forwarded-For"][0], Equals, ip)
-		c.Assert(len(req.Header["X-Request-Id"][0]), Equals, 32)
+		// size of UUID + the 4 hyphens
+		// maybe better to actually parse the UUID back to bytes here?
```

Comment:
I kind of feel like we should parse the UUID here for completeness rather than just counting bytes, thoughts?